src/Block/Service/MenuBlockService.php

```diff
@@ -153,11 +163,7 @@ protected function getFormSettingsKeys()
             'required' => false,
         ];
 
-        $choices = $this->menus;
-
-        if (0 == count($choices)) {
-            $choices = $this->menuRegistry->getAliasNames();
-        }
```

Comment:
Is this BC acceptable?

Reply:
Should be acceptable, because you created a new local `MenuRegistry` and moved all aliases to the registry